Connectors/Rhino/Speckle.Connectors.Rhino7/HostApp/RhinoMaterialManager.cs

```diff
@@ -0,0 +1,147 @@
+  public bool Contains(Material material)
+  {
+    return _renderMaterialCache.TryGetValue(material.Id.ToString(), out SpeckleRenderMaterial _);
+  }
+
+  public BakeResult BakeMaterials(
+    List<SpeckleRenderMaterial> speckleRenderMaterials,
+    string baseLayerName,
+    Action<string, double?>? onOperationProgressed
+  )
+  {
+    var doc = RhinoDoc.ActiveDoc; // POC: too much right now to interface around
+
+    // Keeps track of the incoming SpeckleRenderMaterial application Id and the index of the corresponding Rhino Material in the doc material table
+    Dictionary<string, int> materialIdAndIndexMap = new();
+
+    int count = 0;
+    List<ReceiveConversionResult> conversionResults = new();
+    foreach (SpeckleRenderMaterial speckleRenderMaterial in speckleRenderMaterials)
+    {
+      onOperationProgressed?.Invoke("Converting render materials", (double)++count / speckleRenderMaterials.Count);
+      try
+      {
+        // POC: Currently we're relying on the render material name for identification if it's coming from speckle and from which model; could we do something else?
+        string matName = $"{speckleRenderMaterial.name}-({speckleRenderMaterial.applicationId})-{baseLayerName}";
+        Color diffuse = Color.FromArgb(speckleRenderMaterial.diffuse);
+        Color emissive = Color.FromArgb(speckleRenderMaterial.emissive);
+        double transparency = 1 - speckleRenderMaterial.opacity;
+
+        Material rhinoMaterial =
+          new()
+          {
+            Name = matName,
+            DiffuseColor = diffuse,
+            EmissionColor = emissive,
+            Transparency = transparency
+          };
+
+        int matIndex = doc.Materials.Add(rhinoMaterial);
+
+        // POC: check on matIndex -1, means we haven't created anything - this is most likely an recoverable error at this stage
+        if (matIndex == -1)
+        {
+          throw new ConversionException("Failed to add a material to the document.");
+        }
+
+        if (speckleRenderMaterial.applicationId != null)
+        {
+          materialIdAndIndexMap[speckleRenderMaterial.applicationId] = matIndex;
+        }
+
+        conversionResults.Add(new(Status.SUCCESS, speckleRenderMaterial, matName, "Material"));
+      }
+      catch (Exception ex) when (!ex.IsFatal())
+      {
+        conversionResults.Add(new(Status.ERROR, speckleRenderMaterial, null, null, ex));
+      }
+    }
+
+    return new(materialIdAndIndexMap, conversionResults);
+  }
+
+  /// <summary>
+  /// Removes all materials with a name starting with <paramref name="namePrefix"/> from the active document
+  /// </summary>
+  /// <param name="namePrefix"></param>
+  public void PurgeMaterials(string namePrefix)
+  {
+    var currentDoc = RhinoDoc.ActiveDoc; // POC: too much right now to interface around
+    foreach (Material material in currentDoc.Materials)
+    {
+      if (!material.IsDeleted && material.Name.Contains(namePrefix))
```

Comment:
No action needed: later we should handle these checks with UserDictionaries instead of messing with the name of materials.